warehouse/query-core/src/test/java/datawave/query/util/DateIndexIteratorTest.java

```diff
@@ -0,0 +1,247 @@
+        withExpectedNextSeekCounts(10, 1);
+        drive("20250404", "LOADED");
+    }
+
+    @Test
+    public void testRestrictScanRangeByDate() throws IOException {
+        write("20250401", "LOADED", "20250401", "datatype-a", "FIELD_A", 1);
+        write("20250401", "LOADED", "20250402", "datatype-a", "FIELD_A", 2);
+        write("20250401", "LOADED", "20250403", "datatype-a", "FIELD_A", 3);
+
+        write("20250402", "LOADED", "20250401", "datatype-a", "FIELD_A", 4);
+        write("20250402", "LOADED", "20250402", "datatype-a", "FIELD_A", 5);
+        write("20250402", "LOADED", "20250403", "datatype-a", "FIELD_A", 6);
+
+        write("20250403", "LOADED", "20250401", "datatype-a", "FIELD_A", 7);
+        write("20250403", "LOADED", "20250402", "datatype-a", "FIELD_A", 8);
+        write("20250403", "LOADED", "20250403", "datatype-a", "FIELD_A", 9);
+
+        // full scan range
+        withExpectedNextSeekCounts(10, 1);
+        drive("20250401", "20250403", "LOADED");
+
+        // full scan range, restrict to single day in the column qualifier
+        for (String date : List.of("20250401", "20250402", "20250403")) {
+            withStartEndDate(date, date);
+            withExpectedNextSeekCounts(4, 1);
+            drive("20250401", "20250403", "LOADED");
+        }
+    }
+
+    @Test
+    public void testFieldFilter() throws IOException {
+        write("20250401", "LOADED", "20250401", "datatype-a", "FIELD_A", 1);
+        write("20250401", "LOADED", "20250402", "datatype-a", "FIELD_B", 2);
+        write("20250401", "LOADED", "20250403", "datatype-a", "FIELD_C", 3);
+
+        for (String field : List.of("FIELD_A", "FIELD_B", "FIELD_C")) {
+            withExpectedNextSeekCounts(2, 1);
+            withField(field);
+            drive("20250401", "LOADED");
+        }
+    }
+
+    private void write(String shard, String type, String date, String datatype, String field, int... offsets) {
+        Key key = new Key(shard, type, date + "\u0000" + datatype + "\u0000" + field);
+        BitSet bitset = new BitSet();
+        for (int offset : offsets) {
+            bitset.set(offset);
+        }
+        Value value = new Value(bitset.toByteArray());
+        data.put(key, value);
+    }
+
+    private void drive(String date, String type) throws IOException {
+        drive(date, date, type);
+    }
+
+    private void drive(String start, String stop, String type) throws IOException {
+        Range range = createRangeForDate(start, stop);
+        SortedKeyValueIterator<Key,Value> source = new SortedMapIterator(data);
+
+        Collection<ByteSequence> columnFamily = Collections.singleton(new ArrayByteSequence(type.getBytes()));
+
+        StatDateIndexIterator iter = new StatDateIndexIterator();
+        iter.init(source, options, env);
+        iter.seek(range, columnFamily, true);
+
+        while (iter.hasTop()) {
+            Key key = iter.getTopKey();
+            Value value = iter.getTopValue();
```

Comment:
The variable `value` is never used. Should this be logged below alongside the key?